src/SignalRServiceExtension/TriggerBindings/Executor/SignalRMethodExecutor.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal abstract class SignalRMethodExecutor
+    {
+        protected IRequestResolver Resolver { get; }
+        protected ExecutionContext ExecutionContext { get; }
+
+        protected SignalRMethodExecutor(IRequestResolver resolver, ExecutionContext executionContext)
+        {
+            Resolver = resolver ?? throw new ArgumentNullException(nameof(resolver));
+            ExecutionContext = executionContext ?? throw new ArgumentNullException(nameof(executionContext));
+        }
+
+        public abstract Task<HttpResponseMessage> ExecuteAsync(HttpRequestMessage request);
+
+        protected Task<FunctionResult> ExecuteWithAuthAsync(HttpRequestMessage request, ExecutionContext executor,
+            InvocationContext context, TaskCompletionSource<object> tcs = null)
+        {
+            if (!Resolver.ValidateSignature(request, executor.AccessKey))
+            {
+                //TODO: More detailed exception
+                throw new SignalRTriggerException();
+            }
+
+            return ExecuteAsync(executor.Executor, context, tcs);
+        }
+
+        private async Task<FunctionResult> ExecuteAsync(ITriggeredFunctionExecutor executor, InvocationContext context, TaskCompletionSource<object> tcs)
```

Comment:
>ExecuteAsync [](start = 43, length = 12)

How about rename to `ExcuteAsyncCore`, kind of confusing with the abtract method `ExecuteAsync`.

Reply:
Sure